src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -5572,6 +5776,66 @@ .maxstack  9
             }
         }
 
+        [Fact]
+        public void SpreadElement_12()
+        {
+            string source = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        object x = new[] { 2, 3 };
+                        int[] y = [1, ..x];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics(
+                // (6,25): error CS1579: foreach statement cannot operate on variables of type 'object' because 'object' does not contain a public instance or extension definition for 'GetEnumerator'
```

Comment:
I think people will find this diagnostic confusing, but also this PR is not the place to change it.

Reply:
https://github.com/dotnet/roslyn/issues/69745